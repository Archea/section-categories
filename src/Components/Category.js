import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import { withRouter } from 'react-router'

export class Category extends Component {
  render() {
    return (
      <div className="Category">
        <Link
          to={`/Category/${this.props.Category.id}`}
          style={{ textDecoration: 'none' }}>
          <div>{this.props.Category.name}</div>
        </Link>
      </div>
    )
  }

  _pickCategory = async () => {
    // ... you'll implement this
  }
}

export default withRouter(Category)
