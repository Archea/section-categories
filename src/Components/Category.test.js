import React from 'react'
import ReactDOM from 'react-dom'
import { Category } from './Category'
import renderer from 'react-test-renderer'
import { StaticRouter } from 'react-router'

describe('Category', () => {
  const mocks = {
    Category: {
      id: '1',
      name: 'bar'
    }
  }

  test('it at renders and matches snapshot', () => {
    const context = {}
    const comp = renderer.create(
      <StaticRouter location="/" context={context}>
        <Category {...mocks} />
      </StaticRouter>
    )
    expect(comp).toBeDefined()
    expect(comp.toJSON()).toMatchSnapshot()
  })
})
