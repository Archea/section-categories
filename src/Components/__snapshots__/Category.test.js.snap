// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category it at renders and matches snapshot 1`] = `
<div>
  <a
    href="/Category/1"
    onClick={[Function]}
  >
    <div>
      bar
    </div>
  </a>
</div>
`;
